client/http/http.go

```diff
@@ -7,31 +7,64 @@ import (
 	"context"
 	"io"
 	"net/http"
+	"strconv"
 	"time"
 
+	"github.com/opentracing-contrib/go-stdlib/nethttp"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/prometheus/client_golang/prometheus"
+
 	"github.com/beatlabs/patron/correlation"
 	"github.com/beatlabs/patron/encoding"
 	"github.com/beatlabs/patron/log"
 	"github.com/beatlabs/patron/reliability/circuitbreaker"
 	"github.com/beatlabs/patron/trace"
-	"github.com/opentracing-contrib/go-stdlib/nethttp"
-	opentracing "github.com/opentracing/opentracing-go"
-	"github.com/opentracing/opentracing-go/ext"
 )
 
 const (
 	clientComponent = "http-client"
 )
 
+var (
+	reqTotalMetric   *prometheus.CounterVec
+	reqLatencyMetric *prometheus.HistogramVec
```

Comment:
IIRC Histogram has already a counter included. https://prometheus.io/docs/practices/histograms/#count-and-sum-of-observations

Reply:
Not sure if we can get away with using just a single `HistogramVec`.  Its API returns an `Observer` interface which cannot be directly incremented. 

The `Observer` interface _is_ used by `Summary` though, so it may be possible to replace the two counters with a single `SummaryVec`? Not sure if this would give us the same end-result.

Note that we've used the same approach, counter and histogram, for the [gRPC component](https://github.com/beatlabs/patron/blob/master/component/grpc/observability.go).